frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,201 @@
+    }
+
+    onChange = key => {
+      const { name, kindLabel, accessMode } = _.get(this.state, ['items', key], {});
+      this.setState({
+        selectedKey: key,
+        title: this.getTitle(kindLabel, name),
+      });
+      this.props.onChange(name, accessMode);
+    };
+
+    render() {
+      const { placeholder, id, loaded } = this.props;
+      const items = {};
+      const sortedItems = _.keys(this.state.items).sort();
+      _.each(
+        this.state.items,
+        (props, key) =>
+          (items[key] = <StorageClassDropdownEntry {...props} />)
+      );
+
+      const { selectedKey, defaultClass } = this.state;
+
+      const Component = <Dropdown
+        autocompleteFilter={this.autocompleteFilter}
+        autocompletePlaceholder={placeholder}
+        items={items}
+        sortedItemKeys={sortedItems}
+        selectedKey={selectedKey}
+        title={this.state.title}
+        onChange={this.onChange}
+        id={id}
+        menuClassName="dropdown-menu--text-wrap"
+      />;
+      // Only show the dropdown if 'no storage class' is not the only option which depends on defaultClass
+      const itemsAvailableToShow = (_.size(items)>1 && !defaultClass) || (_.size(items)>0 && defaultClass);
+      const helpMes = defaultClass ? `Default storage class ${defaultClass} is set so that will be used if no other storage class selected` : 'Optional storage class for the new claim.';
+      return <React.Fragment>
+        {loaded && itemsAvailableToShow &&
+          <div>
+            <label className="control-label" htmlFor="storageclass-dropdown">
+            Storage Class
+            </label>
+            {Component}
+            <p className="help-block" id="storageclass-dropdown-help">
+              {helpMes}
+            </p>
+          </div>
+        }
+      </React.Fragment>;
+    }
+}
+
+export const StorageClassDropdown = props => {
+  return <Firehose resources={[{ kind: 'StorageClass', prop: 'StorageClass', isList: true }]}>
+    <StorageClassDropdown_ {...props} />
+  </Firehose>;
+};
+
+const StorageClassDropdownEntry = props => {
+  const storageClassProperties = [props.default ? ' (default)' : '', props.description, props.accessMode, props.provisioner, props.type, props.zone];
+  const storageClassDescriptionLine = storageClassProperties.filter(Boolean).join(' | ');
+  return <span>
+    {props.kindLabel ? <ResourceName kind={props.kindLabel} name={props.name} /> : <span >{props.name}</span>}
+    <span className="text-muted co-form-subsection"> {storageClassDescriptionLine}</span>
+  </span>;
+};
+
+export type StorageClassDropdownState = {
+    items: any;
+    name: string;
+    selectedKey: string;
+    title: any;
+    defaultClass: string;
+};
+
+export type StorageClassDropdownProps = {
+    id?: string;
+    loaded?: boolean;
+    name: string;
+    onChange: any;
+    placeholder?: string;
+    resources: any;
+    selectedKey: string;
+    selectedKeyKind: any;
```

Comment:
It's not clear to me why `selectedKey` and `selectedKeyKind` are passed in as props. You know the kind, and selected key should just be passed `onChange`.

We can always add this later if we need to preselect a value for edit (although I'm not sure if storage class is mutable for a PVC).

I realize this is how the other dropdown worked, but it's generally a bad practice to have the same values in props and state.